File: Data Structures & Algorithms/top-k-elements-in-list/submission-1.py
class Solution:
    def topKFrequent(self, nums: List[int], k: int) -> List[int]:
        counts = {}
        for n in nums:
            if n in counts:
                counts[n] += 1
            else:
                counts[n] = 1
        
        buckets = [[] for _ in range(len(nums) + 1)]
        for num,count in counts.items():
            buckets[count].append(num)
        
        res = []
        for nums in reversed(buckets):
            for n in nums:
                res.append(n)
                if len(res) == k:
                    return res

File: Data Structures & Algorithms/majority-element/submission-1.cpp
class Solution {
public:
    int majorityElement(vector<int>& nums) {
        int res = 0, count = 0;
        for(int num : nums){
            if(count == 0){
                res = num;
            }
            res == num ? count++ : count--;
        }
        return res;
    }
};